locals.tf

```diff
@@ -11,11 +11,15 @@
 // limitations under the License.
 
 locals {
-  subnet_names_prefixes = zipmap(var.subnet_names, var.subnet_prefixes)
-
-  azurerm_subnets = var.use_for_each ? [for s in azurerm_subnet.subnet_for_each : s] : [for s in azurerm_subnet.subnet_count : s]
   azurerm_subnets_name_id_map = {
-    for index, subnet in local.azurerm_subnets :
+    for subnet in azurerm_subnet.subnet :
     subnet.name => subnet.id
   }
+
+  default_tags = {
+    provisioner   = "terraform"
+    resource_name = var.vnet_name
```

Comment:
just observed this now. although the resource_name has been added in the primitive module but still we may have to override it sometimes in the wrappers. The direction from Matt on the resource_name was that it should always contain the name generated by naming module `standard`. For some resources like key-vault or storage account where because of length constraint we use shorter names for the actual name, in those cases still the resource_name tag should be the `module.names["storage"].standard`

Reply:
1. for primitive I think it doesnt matter if this tag is set IMO. Because its very unlikely that we are going to use a primitive to provision the actual infrastructure anywhere. And even if we do, then without the naming module, its upto the discretion of the user to put the name in correct format. so even if this tag is added in the primitive module, we would certainly override it in the wrapper which is wrapping this primitive module. 
2. For examples, we should not even bother to have these tags. as the examples are meant to be for demo purposes and would be short lived